### PR TITLE
New resource - RDS Parameter Group

### PR DESCRIPTION
**Pull request by scotwk**
#1276 - adding new resources for RDS Parameter Groups and RDS Cluster Parameter Groups.

### CODE REVIEW

**Comment by scotwk** [comment]
@kapilt - A few questions in #1276 about whether more functionality is wanted, but otherwise good to go.

File: c7n/resources/rdsparamgroup.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+from c7n.actions import ActionRegistry, BaseAction
+from c7n.filters import FilterRegistry
+from c7n.manager import resources
+from c7n.query import QueryResourceManager
+from c7n.utils import (type_schema, local_session)
+
+log = logging.getLogger('custodian.rds-param-group')
+
+pg_filters = FilterRegistry('rds-param-groups.filters')
```
Comment (kapilt) [inline c7n/resources/rdsparamgroup.py]: this should match the resource type name i think (note pluralization). also you don't have to specify these out of band, the metaclass will auto create them for you though its more typing with the resource manager class name as a prefix.

File: c7n/resources/rdsparamgroup.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            params.append({
+                'ParameterName': param['name'],
+                'ParameterValue': param['value'],
+                'ApplyMethod': param.get('apply-method', 'immediate'),
+            })
+
+        # Can only do 20 elements at a time per docs, so if we have more than that we will
+        # break it into multiple requests: https://goo.gl/Z6oGNv
+        params = [iter(params)] * 20
+        params = itertools.izip_longest(*params, fillvalue={})
```
Comment (kapilt) [inline c7n/resources/rdsparamgroup.py]: we have a chunks method for doing this. aka
`for param_set  in chunks(params, 20)`

File: c7n/resources/rdsparamgroup.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+        for param_set in params:
+            for param_group in param_groups:
+                name = self.get_pg_name(param_group)
+                try:
+                    self.do_modify(client, name, param_set)
```
Comment (kapilt) [inline c7n/resources/rdsparamgroup.py]: so we also need to check the value we're setting to is actually different then the value that's already there, if its not aka we have no modifications for the set of param changes, we shouldn't do the api call.

**Comment by kapilt** [comment]
flake8 reports `c7n/resources/rdsparamgroup.py:178:20: F821 undefined name 'e'`

**Comment by scotwk** [comment]
@kapilt - fixed

**Pull request merged**